liquidity-pool-changes/src/agent.js

```diff
@@ -0,0 +1,219 @@
+      erc20Abi,
+      provider,
+    );
+
+    // get amount of eth in the pool if applicable (if the pool is using WETH, then this value is 0)
+    const poolEthBalance = await provider.getBalance(poolContract.address);
+
+    // get the # amount of each tokens in the liquidity pool
+    const tokenBalance0 = await token0Contract.balanceOf(poolContract.address);
+    const tokenBalance1 = await token1Contract.balanceOf(poolContract.address);
+
+    // convert # amount of each tokens in the liquidity pool from ethers BigNumber to BigNumber.js
+    const tokenBalance0BN = new BigNumber(tokenBalance0.toString());
+    const tokenBalance1BN = new BigNumber(tokenBalance1.toString());
+    const poolEthBalanceBN = new BigNumber(poolEthBalance.toString());
+
+    // this returns the token prices in usd of each token (without accounting for decimals)
+    let tokenPrices;
+    try {
+      tokenPrices = await getTokenPrices(token0Address, token1Address);
+    } catch (error) {
+      // if coingecko call fails
+      return findings;
+    }
+
+    // multiply the amount of tokens in the pool by usd value, adjusting for decimals
+    let token0ValueUSD;
+    let token1ValueUSD;
+    let currentLiquidity;
+
+    if (tokenBalance0BN.gt(0)) {
+      token0ValueUSD = await getTokenUSDValue(
+        tokenBalance0BN,
+        tokenPrices.token0Price,
+        token0Address,
+        provider,
+      );
+    }
+
+    if (tokenBalance1BN.gt(0)) {
+      token1ValueUSD = await getTokenUSDValue(
+        tokenBalance1BN,
+        tokenPrices.token1Price,
+        token1Address,
+        provider,
+      );
+    }
+
+    // calculate the total liquidity value of the pool at current block
+    // @dev for sanity check, liquidity should be close to the one here for usdc/eth pool https://etherscan.io/address/0x8ad599c3a0ff1de082011efddc58f1908eb6e6d8
+    /* eslint no-unused-expressions: ["error", {"allowTernary": true }] */
+    if (token0ValueUSD !== undefined && token1ValueUSD !== undefined) {
+      counter === 0
+        ? (previousLiquidity = token0ValueUSD.plus(token1ValueUSD))
+        : (currentLiquidity = token0ValueUSD.plus(token1ValueUSD));
+    } else if (token0ValueUSD !== undefined && poolEthBalanceBN.gt(0)) {
+      counter === 0
+        ? (previousLiquidity = token0ValueUSD.plus(poolEthBalanceBN))
+        : (currentLiquidity = token0ValueUSD.plus(poolEthBalanceBN));
+    } else if (token1ValueUSD !== undefined && poolEthBalanceBN.gt(0)) {
+      counter === 0
+        ? (previousLiquidity = token1ValueUSD.plus(poolEthBalanceBN))
+        : (currentLiquidity = token1ValueUSD.plus(poolEthBalanceBN));
+    }
+    /* eslint no-unused-expressions: ["error", { "allowTernary": true }] */
+
+    // increment counter to signal that the agent has already ran on a previous block before
+    counter = 1;
+
+    // return no findings the first time this agent is ran
+    if (currentLiquidity === undefined) {
+      return findings;
+    }
```

Comment:
I feel like this should use rolling math and have a configurable number of previous blocks rather than just one